lib/src/facts/aix/operating_system_resolver.cc

```diff
@@ -0,0 +1,68 @@
+#include <internal/facts/aix/operating_system_resolver.hpp>
+#include <internal/util/regex.hpp>
+#include <facter/facts/os.hpp>
+#include <facter/execution/execution.hpp>
+#include <facter/util/file.hpp>
+
+#include <sstream>
+#include <algorithm>
+
+using namespace std;
+using namespace facter::util;
+namespace execution = facter::execution;
+
+string getattr(string object, string field)
+{
+    string result;
+
+    // To get architecture, can/should we use getconf KERNEL_BITMODE, getconf HARDWARE_BITMODE?
+    execution::each_line(
+        "/usr/sbin/lsattr", {"-El", object, "-a", field},
+        [&](string& line) {
+            if (!line.empty()) {
+                vector<string> tokens;
+                istringstream iss(line);
+                // istream_iterator is a great way to tokenize your life
+                copy(istream_iterator<string>(iss),
+                     istream_iterator<string>(),
+                     back_inserter(tokens));
+                result = tokens[1];
+                return false;
+            }
+            return true;
+        },
+        nullptr,
+        0);
+
+    return result;
+}
+
+namespace facter { namespace facts { namespace aix {
+
+    operating_system_resolver::data operating_system_resolver::collect_data(collection& facts)
+    {
+        // Default to the base implementation
+        auto result = posix::operating_system_resolver::collect_data(facts);
+        result.architecture = getattr("proc0", "type");
+        result.hardware = getattr("sys0", "modelname");
+
+        // Get OS release stuff
+        static boost::regex regexp_release("(\\d+)\\.(\\d+\\.\\d+\\.\\d+)");
```

Comment:
@branan did we decide to just report the `6.1` from the `6.1.7.15` found in `aix_release.level`? If so, we can probably just do something like:

    static boost::regex regexp_release("(\\d+\\.\\d+)\\.\\d+\\.\\d+");
        file::each_line("/usr/lpp/bos/aix_release.level", [&](string& line) {
            string full_release;
            if (re_search(line, regexp_release, &full_release)) {
                result.release = full_release;
                return false;
            }
            return true;
        });
        return result;

No need to concatenate `major + '.' + minor`, as those get parsed in the `parse_release` function below.